src/ratchet.rs

```diff
@@ -65,16 +66,16 @@ impl Ratchet {
     /// ```
     /// use skip_ratchet::Ratchet;
     ///
-    /// let ratchet = Ratchet::new();
+    /// let ratchet = Ratchet::new(&mut rand::thread_rng());
     /// ```
-    pub fn new() -> Self {
+    pub fn new(rng: &mut impl RngCore) -> Self {
```

Comment:
```suggestion
    pub from_rng<T: RngCore + CryptoRng>(mut csprng: T) -> Self {
```

fwiw - I would trait bound it and `new()` is problematic - I've been deprecating elsewhere this pattern to as `from_x` is more explicit and so I am able to gate them as optionals.
Problem with `new()` is to decide what it does - In addition `rand_core` is not stable API so stabilising API without SemVer exceptions to rand_core (which potentially may be good to have behind feature) - `new()` imo should have no inputs in constructor that hides everything behind as impl detail.